### PR TITLE
mbedtls: make `PSA_WANT_KEY_TYPE_xxx_KEY_PAIR_BASIC` promptless

### PR DESCRIPTION
**Pull request by valeriosetti**
`PSA_WANT_KEY_TYPE_xxx_KEY_PAIR_BASIC` will be selected by any of:

- `PSA_WANT_KEY_TYPE_xxx_KEY_PAIR_IMPORT`
- `PSA_WANT_KEY_TYPE_xxx_KEY_PAIR_EXPORT`
- `PSA_WANT_KEY_TYPE_xxx_KEY_PAIR_GENERATE`
- `PSA_WANT_KEY_TYPE_xxx_KEY_PAIR_DERIVE`. 

This mechanism mimic what Mbed TLS already does internally with the corresponding build symbols.

### CODE REVIEW

File: modules/mbedtls/Kconfig.psa [hunk 1]
```diff
@@ -301,7 +301,7 @@ config PSA_WANT_KEY_TYPE_RSA_PUBLIC_KEY
 	default y if PSA_CRYPTO_ENABLE_ALL
 
 config PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_BASIC
-	bool "PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_BASIC" if !MBEDTLS_PROMPTLESS
+	bool
 	default y if PSA_CRYPTO_ENABLE_ALL
```
Comment (tomi-font) [inline modules/mbedtls/Kconfig.psa:305]: Remove those `PSA_WANT_KEY_TYPE_*_KEY_PAIR_BASIC` completely from this file as you are adding them to `Kconfig.psa.logic`?

File: modules/mbedtls/Kconfig.psa.logic [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+# This file extends Kconfig.psa (which is automatically generated) by adding
+# some logic between PSA_WANT symbols.
+
+config PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_BASIC
+	default y
+	depends on PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_IMPORT || \
+		   PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_EXPORT || \
+		   PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_GENERATE || \
+		   PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_DERIVE
```
Comment (tomi-font) [inline modules/mbedtls/Kconfig.psa.logic:12]: In your commit message you say that the features select `*_BASIC`, but you are doing it the other way around in practice (`*_BASIC` auto-enable themselves).

**Review by tomi-font — CHANGES_REQUESTED** [review]

File: modules/mbedtls/create_psa_files.py [hunk 3]
```diff
@@ -57,6 +57,20 @@
...
 H_FOOTER="\n#endif /* CONFIG_PSA_H */\n"
 
+# In Mbed TLS the PSA_WANT_KEY_TYPE_[ECC|RSA|DH]_KEY_PAIR_BASIC build symbols
+# are automatically enabled whenever any other _IMPORT, _EXPORT, _GENERATE or
+# _DERIVE feature is set for the same key type
+# (see "modules/crypto/mbedtls/include/psa/crypto_adjust_config_key_pair_types.h").
+# Therefore we mimic the same pattern with Kconfigs as follows:
+# - do not add _BASIC Kconfigs to the automatic generated file (KCONFIG_PATH);
+# - add _BASIC Kconfigs to Kconfig.psa.logic and let them "default y" as soon as
+#   any other _IMPORT, _EXPORT, _GENERATE or _DERIVE Kconfigs are enabled.
```
Comment (tomi-font) [inline modules/mbedtls/create_psa_files.py:67]: Not sure such a length comment was needed (could have been more generic), but it's good as is to me. :slightly_smiling_face:

**Review by tomi-font — DISMISSED** [review]

**Review by tomi-font — DISMISSED** [review]

**Review by tomi-font — APPROVED** [review]

**Review by frkv — APPROVED** [review]
LGTM

**Review by carlescufi — APPROVED** [review]

**Review by ceolin — APPROVED** [review]

**Pull request merged**